#514 deprecate unused meta packages

mjcarsjens opened the pull request:
Deprecated unused meta-packages, and adjusted denhaag-component-library package accordingly.

**Closing issues**
closes #514

bddjong commented on packages/denhaag-component-library/package.json:
```diff
@@ -22,13 +22,31 @@
-    "@gemeente-denhaag/navigation": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/alert": "^0.1.1-alpha.20",
+    "@gemeente-denhaag/avatar": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/badge-counter": "^0.1.1-alpha.50",
+    "@gemeente-denhaag/divider": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/list": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/typography": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/button": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/checkbox": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/formcontrol": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/formcontrollabel": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/formgroup": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/iconbutton": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/inputlabel": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/radio": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/select": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/switch": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/textfield": "^0.2.3-alpha.50",
     "@gemeente-denhaag/pickers": "^0.2.3-alpha.50",
-    "@gemeente-denhaag/stylesprovider": "^0.1.1-alpha.50",
-    "@gemeente-denhaag/surfaces": "^0.2.3-alpha.50"
+    "@gemeente-denhaag/link": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/menu": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/tab": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/timeline": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/accordion": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/card": "^0.2.3-alpha.50",
+    "@gemeente-denhaag/stylesprovider": "^0.1.1-alpha.50"
```
Can we make sure these are sorted alphabetically? npm-package-json-lint should be warning us about that with the following rule https://npmpackagejsonlint.org/docs/en/rules/dependencies/prefer-alphabetical-dependencies#example-npmpackagejsonlintrc-configuration

bddjong commented on packages/denhaag-component-library/src/index.tsx:
```diff
@@ -1,6 +1,27 @@
-export * from '@gemeente-denhaag/layout';
-export * from '@gemeente-denhaag/navigation';
-export * from '@gemeente-denhaag/surfaces';
+export * from '@gemeente-denhaag/baseprops';
+export * from '@gemeente-denhaag/alert';
+export * from '@gemeente-denhaag/avatar';
+export * from '@gemeente-denhaag/badge-counter';
+export * from '@gemeente-denhaag/divider';
+export * from '@gemeente-denhaag/list';
+export * from '@gemeente-denhaag/typography';
+export * from '@gemeente-denhaag/button';
+export * from '@gemeente-denhaag/checkbox';
+export * from '@gemeente-denhaag/formcontrol';
+export * from '@gemeente-denhaag/formcontrollabel';
+export * from '@gemeente-denhaag/formgroup';
+export * from '@gemeente-denhaag/iconbutton';
+export * from '@gemeente-denhaag/inputlabel';
+export * from '@gemeente-denhaag/pickers';
+export * from '@gemeente-denhaag/pickersutilsprovider';
+export * from '@gemeente-denhaag/radio';
+export * from '@gemeente-denhaag/select';
+export * from '@gemeente-denhaag/switch';
+export * from '@gemeente-denhaag/textfield';
+export * from '@gemeente-denhaag/menu';
+export * from '@gemeente-denhaag/timeline';
+export * from '@gemeente-denhaag/tab';
+export * from '@gemeente-denhaag/link';
+export * from '@gemeente-denhaag/accordion';
+export * from '@gemeente-denhaag/card';
```
ditto

bddjong reviewed (approved):
Small comments about sorting. Also, we may want to rename the `denhaag-component-library` folder to just `component-library` if you prefer a shorter name. Although I think changing the npm package name is probably not such a good idea.

mjcarsjens commented:
@bddjong changed order around as requested, and added the packagejson lint rule. 

Also changed around the linting setup a bit so folders other than `components` also get linted. 
For stylelint I ran into some issues using `*/**/*.css`, even with correct `.stylelintignore` settings, so I had to resort to specifying the `components`, `proprietary` and `packages` folders here in the command.

mjcarsjens commented:
> Small comments about sorting. Also, we may want to rename the `denhaag-component-library` folder to just `component-library` if you prefer a shorter name. Although I think changing the npm package name is probably not such a good idea.

Not sure about this one. Agree we should not just go around renaming npm packages, but for consistensies sake I would prefer the folder name to correspond to the package name, even though I agree the name `component-library` would be nicer.

Robbert commented:
For the example repo I dropped the `library` entirely, and I would just have `components` / `components-react` / `components-css` / `web-components-react`, et cetera: https://github.com/nl-design-system/example/tree/main/packages

Also more in line with how the directory in the repo itself is called. I can totally rename the one from Utrecht as well. Better now than later.
